service/api_logic/news_logic.py

```diff
@@ -20,24 +19,29 @@ def fetch_news(session, order_by: ClauseElement = None, limit: int = None, filte
 
 
 def get_news_by_count(count: int, session):
-    news = fetch_news(session, order_by=News.save_at.desc(), limit=count)
-    return json_news(news)
+    try:
+        news = fetch_news(session, order_by=News.save_at.desc(), limit=count)
+        return json_news(news)
+    except Exception:
+        raise Exception
 
 
 def get_latest_sport_news(count: int, sport_name: str, session):
     try:
         sport = get_sport_by_name(session, sport_name)
     except SportNotFoundError as e:
-        print_error_message({"error": e.message})
         return get_error_response({"error": e.message },404)
     filters = [News.sport_id == sport.sport_id]
     news = fetch_news(session, order_by=News.save_at.desc(), limit=count, filters=filters)
     return json_news(news)
 
 
 def get_popular_news(count: int, session):
-    news = fetch_news(session, order_by=News.interest_rate.desc(), limit=count)
-    return json_news(news)
+    try:
+        news = fetch_news(session, order_by=News.interest_rate.desc(), limit=count)
+        return json_news(news)
+    except Exception:
```

Comment:
except-raise again

Also, catching "Exception" (base class) here is meaningless, since we don't know how to handle it. It basically means "Something wrong happened" so it's better moved to API's endpoints.

Here, on the other hand, we should only catch exceptions we expect (if any). For instance, let's say you have an algorithm that traverses a list of items by their indexes. You expect that algorithm might not work in certain cases and Some IndexOutOfRange exception might occur. In this case we will catch that paticular exception (try-except IndexOutOfRange as e) and do some custom logic if required or use another algorithm - whatever. Thing is, that we know that this exception might happen and we know how to deal with it - it does not break our logic. Such exceptions are worth catching here, otherwise - catch Exception on the endpoint 

FYI:
by re-raising this Exception you not only allow your app to crush, but also have a good chance of removing original exception's message